File: HackerRank/Python/Easy/E0065.py
# Problem Statement: https://www.hackerrank.com/challenges/re-start-re-end/problem

import re

S, k = input(), input()
matches = [(match.start(1), match.end(1) - 1) for match in re.finditer(f'(?=({k}))', S)]
print(*matches if matches else [(-1, -1)], sep='\n')

File: HackerRank/Algorithms/Easy/E0065.cpp
/*
Problem Statement: https://www.hackerrank.com/challenges/two-strings/problem
*/

#include <iostream>
#include <string>
#include <unordered_set>

using namespace std;

string twoStrings(string s1, string s2) {
	unordered_set<char> s(s1.begin(), s1.end());
	for(char &c : s2)
		if(s.find(c) != s.end())
			return "YES";
	return "NO";
}

int main()
{
	int n;
	string s1, s2;
	cin>>n;
	cin.ignore();
	for(int i=1 ; i<=n ; i++){
		getline(cin, s1);
		getline(cin, s2);
		cout<<twoStrings(s1, s2)<<endl;
	}
	return 0;
}